modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataKeysDelegate.java

```diff
@@ -84,12 +86,11 @@
 
     try {
       final Map<Integer, ParameterModel> partsByOrder = getPartOrderMapping(keyParts);
-      Set<MetadataKey> metadataKeys;
-      if (keyResolver instanceof PartialTypeKeysResolver && hasInitialLevel(partialKey, partsByOrder, reflectionCache)) {
-        metadataKeys = singleton(((PartialTypeKeysResolver) keyResolver).resolveChilds(context, partialKey));
-      } else {
-        metadataKeys = keyResolver.getKeys(context);
-      }
+      Set<MetadataKey> metadataKeys = context instanceof ConnectionProviderAwareMetadataContext
+          ? getWithTokenRefreshIfNecessary(((ConnectionProviderAwareMetadataContext) context).getConnectionProvider()
```

Comment:
this ternary construct is all over the place. Generalize it more.

Reply:
Do you mean like having a util method that receives the context and a supplier and have this instanceof logic there?